codex/utils/iter.nim

```diff
@@ -89,6 +90,36 @@ proc new*[T](_: type Iter[T], items: seq[T]): Iter[T] =
 
   Iter[int].new(0 ..< items.len).map((i: int) => items[i])
 
+proc new*[T](_: type Iter[T], iter: Iterable[T]): Iter[T] =
+  ## Creates a new Iter from an iterator
+  ##
+  var nextOrErr: Option[Result[T, ref CatchableError]]
```

Comment:
Perhaps we can just use `?!` instead of `Result` here. `?!` is already defined as `Result[T, ref CatchableError]`